src/source/Rtp/Codecs/RtpH265Payloader.c

```diff
@@ -0,0 +1,269 @@
+    BOOL sizeCalculationOnly = (payloadBuffer == NULL);
+    PayloadArray payloadArray;
+
+    CHK(nalus != NULL && pPayloadSubLenSize != NULL && pPayloadLength != NULL && (sizeCalculationOnly || pPayloadSubLength != NULL), STATUS_NULL_ARG);
+    CHK(mtu > FU_HEADER_SIZE, STATUS_RTP_INPUT_MTU_TOO_SMALL);
+
+    if (sizeCalculationOnly) {
+        payloadArray.payloadLength = 0;
+        payloadArray.payloadSubLenSize = 0;
+        payloadArray.maxPayloadLength = 0;
+        payloadArray.maxPayloadSubLenSize = 0;
+    } else {
+        payloadArray.payloadLength = *pPayloadLength;
+        payloadArray.payloadSubLenSize = *pPayloadSubLenSize;
+        payloadArray.maxPayloadLength = *pPayloadLength;
+        payloadArray.maxPayloadSubLenSize = *pPayloadSubLenSize;
+    }
+    payloadArray.payloadBuffer = payloadBuffer;
+    payloadArray.payloadSubLength = pPayloadSubLength;
+
+    do {
+        CHK_STATUS(getNextNaluLengthH265(curPtrInNalus, remainNalusLength, &startIndex, &nextNaluLength));
+
+        curPtrInNalus += startIndex;
+
+        remainNalusLength -= startIndex;
+
+        CHK(remainNalusLength != 0, retStatus);
+
+        if (sizeCalculationOnly) {
+            CHK_STATUS(createPayloadFromNaluH265(mtu, curPtrInNalus, nextNaluLength, NULL, &singlePayloadLength, &singlePayloadSubLenSize));
+            payloadArray.payloadLength += singlePayloadLength;
+            payloadArray.payloadSubLenSize += singlePayloadSubLenSize;
+        } else {
+            CHK_STATUS(createPayloadFromNaluH265(mtu, curPtrInNalus, nextNaluLength, &payloadArray, &singlePayloadLength, &singlePayloadSubLenSize));
+            payloadArray.payloadBuffer += singlePayloadLength;
+            payloadArray.payloadSubLength += singlePayloadSubLenSize;
+            payloadArray.maxPayloadLength -= singlePayloadLength;
+            payloadArray.maxPayloadSubLenSize -= singlePayloadSubLenSize;
+        }
+
+        remainNalusLength -= nextNaluLength;
+        curPtrInNalus += nextNaluLength;
+    } while (remainNalusLength != 0);
+
+CleanUp:
+    if (STATUS_FAILED(retStatus) && sizeCalculationOnly) {
+        payloadArray.payloadLength = 0;
+        payloadArray.payloadSubLenSize = 0;
+    }
+
+    if (pPayloadSubLenSize != NULL && pPayloadLength != NULL) {
+        *pPayloadLength = payloadArray.payloadLength;
+        *pPayloadSubLenSize = payloadArray.payloadSubLenSize;
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS getNextNaluLengthH265(PBYTE nalus, UINT32 nalusLength, PUINT32 pStart, PUINT32 pNaluLength)
+{
+    ENTERS();
+
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT32 zeroCount = 0;
+    UINT32 offset;
+    BOOL naluFound = FALSE;
+    PBYTE pCurrent = NULL;
+
+    CHK(nalus != NULL && pStart != NULL && pNaluLength != NULL, STATUS_NULL_ARG);
+
+    // Annex-B Nalu will have 0x000000001 or 0x000001 start code, at most 4 bytes
+    for (offset = 0; offset < 4 && offset < nalusLength && nalus[offset] == 0; offset++)
+        ;
```

Comment:
Can we use a while loop here, with our current code stye we forbid a for loop with no { }.  I would suggest re-writing this something like:

Initialize offset to zero at the start of the function, then:

```
while(offset < 4 && offset < nalusLength && nalus[offset] == 0) {
    offset++;
}
```

Reply:
This was copied over from the H264 payloader. The functions `getNextNaluLength` and `getNextNaluLengthH265` are identical. Would you recommend replacing it in both?

Reply:
Yes

Reply:
Done!